core/ops.rs

```diff
@@ -20,19 +22,83 @@ use std::ops::DerefMut;
 use std::pin::Pin;
 use std::rc::Rc;
 
-pub type OpAsyncFuture = Pin<Box<dyn Future<Output = Box<[u8]>>>>;
-pub type OpFn = dyn Fn(Rc<RefCell<OpState>>, BufVec) -> Op + 'static;
+pub use erased_serde::Serialize as Serializable;
+pub type PromiseId = u64;
+pub type OpAsyncFuture = Pin<Box<dyn Future<Output = OpResponse>>>;
+pub type OpFn =
+  dyn Fn(Rc<RefCell<OpState>>, OpPayload, Option<ZeroCopyBuf>) -> Op + 'static;
 pub type OpId = usize;
 
+pub struct OpPayload<'a, 'b, 'c> {
+  pub(crate) scope: Option<&'a mut v8::HandleScope<'b>>,
+  pub(crate) value: Option<v8::Local<'c, v8::Value>>,
+}
+
+impl<'a, 'b, 'c> OpPayload<'a, 'b, 'c> {
+  pub fn new(
+    scope: &'a mut v8::HandleScope<'b>,
+    value: v8::Local<'c, v8::Value>,
+  ) -> Self {
+    Self {
+      scope: Some(scope),
+      value: Some(value),
+    }
+  }
+
+  pub fn empty() -> Self {
+    Self {
+      scope: None,
+      value: None,
+    }
+  }
+
+  pub fn deserialize<T: DeserializeOwned>(self) -> Result<T, AnyError> {
+    serde_v8::from_v8(self.scope.unwrap(), self.value.unwrap())
```

Comment:
Is it expected that this function panics for `OpPayload::empty()`? A bit surpising

Reply:
In production these should never be `None`, we only use `OpPayload::empty()` for the op-table test, so we don't have to spin up a v8 isolate just to test that

Reply:
So maybe add `#[test]` attribute to `OpPayload::empty`